lib/rmg/CRmgTemplateZone.cpp

```diff
@@ -1174,10 +1174,10 @@ void CRmgTemplateZone::initTownType ()
 		if (playerInfo.canAnyonePlay())
 		{
 			player = PlayerColor(player_id);
-			townType = gen->mapGenOptions->getPlayersSettings().find(player)->second.getStartingTown();
+			townType = gen->getMapGenOptions().getPlayersSettings().find(player)->second.getStartingTown();
 
 			if (townType == CMapGenOptions::CPlayerSettings::RANDOM_TOWN)
-				randomizeTownType();
+				randomizeTownType(true);
```

Comment:
Why do we set true only here. We have 2 more calls to this function?

Reply:
I do not know HOMM3 specifics regarding map generation, so just ask.

Reply:
Line 1185

Reply:
Ok, understood